tests/baselines/reference/typeFromPropertyAssignment29.js

```diff
@@ -188,13 +188,13 @@ declare const ExpandoExpr: {
         y: string;
         x?: undefined;
     };
-    m(n: number): number;
+    m: (n: number) => number;

```

Comment:
Yes, the change is intended: expando properties on a function shouldn't give you static methods (like they used to). They give you exports from a merged namespace. If you want bivariance you'll have to use explicit method syntax. 
This is easily available in Typescript, and I don't expect to see a lot of `--strict` adoption in JS*, so I think this is the right approach.

* Except by people willing to rewrite their entire source a la webpack.

Reply:
But prototype assignments still get you methods (right)? Since prototype assignments are instance methods. I guess because that's expected, the expanding assignments being different feels a bit odd.

Reply:
Yeah, this is tricky. Basically, we have to choose how we'd like to model property-assignment declarations. Like class statics:

```ts
class Ex {
    static m = 1
}
namespace Ex {
    export var m = 2
}
```

Or namespace merges:

```ts
namespace Mr {
    export var n = 3
}
namespace Mr {
    export var n = 4
}
```

Taken on their own, I think property-assignment declarations are actually more like namespace merges. And this model allows merges of properties of the same name, which is .. nice? I guess? You definitely can't merge `Ex.m` in the first example.

On the other hand, classic JS often uses prototype-property- and property-assignment declarations to model classic OO:

```js
function Class() {
}
Class.prototype.method = ...
Class.staticMethod = ...
```

If you are around tomorrow let's take some time to discuss this in person and see if we can decide which way to go.